File: src/main/java/gg/mineral/practice/inventory/menus/SelectEventMenu.java
package gg.mineral.practice.inventory.menus;

import gg.mineral.practice.events.Event;

import gg.mineral.practice.inventory.ClickCancelled;
import gg.mineral.practice.inventory.PracticeMenu;
import gg.mineral.practice.managers.GametypeManager;
import gg.mineral.practice.util.items.ItemBuilder;
import gg.mineral.practice.util.messages.CC;
import gg.mineral.practice.util.messages.impl.ErrorMessages;
import lombok.val;

@ClickCancelled(true)
public class SelectEventMenu extends PracticeMenu {
    @Override
    public void update() {
        for (val g : GametypeManager.getGametypes().values()) {
            if (!g.isEvent())
                continue;

            val item = new ItemBuilder(g.getDisplayItem())
                    .name(CC.SECONDARY + CC.B
                            + (g.isInCategory() ? g.getCategoryName() + g.getDisplayName() : g.getDisplayName()))
                    .lore(CC.ACCENT + "Click to start event.")
                    .build();

            add(item, interaction -> {
                viewer.getPlayer().closeInventory();

                viewer.getDuelSettings().setGametype(g);

                if (g.getEventArenaId() == -1) {
                    viewer.message(ErrorMessages.ARENA_NOT_FOUND);
                    return;
                }

                Event event = new Event(viewer, g.getEventArenaId());
                event.start();
            });
        }
    }

    @Override
    public String getTitle() {
        return CC.BLUE + "Select Event";
    }

    @Override
    public boolean shouldUpdate() {
        return true;
    }
}

File: src/main/kotlin/gg/mineral/practice/inventory/menus/SelectEventMenu.kt
package gg.mineral.practice.inventory.menus

import gg.mineral.practice.events.Event
import gg.mineral.practice.inventory.ClickCancelled
import gg.mineral.practice.inventory.PracticeMenu
import gg.mineral.practice.managers.GametypeManager.gametypes
import gg.mineral.practice.util.items.ItemBuilder
import gg.mineral.practice.util.messages.CC
import gg.mineral.practice.util.messages.impl.ErrorMessages

@ClickCancelled(true)
class SelectEventMenu : PracticeMenu() {
    override fun update() {
        for (g in gametypes.values) {
            if (!g.event) continue

            val item = ItemBuilder(g.displayItem)
                .name(
                    (CC.SECONDARY + CC.B
                            + (if (g.inCategory) g.categoryName + g.displayName else g.displayName))
                )
                .lore(CC.ACCENT + "Click to start event.")
                .build()

            add(item) {
                viewer.player.closeInventory()
                viewer.duelSettings.gametype = g

                if (g.eventArenaId.toInt() == -1) {
                    viewer.message(ErrorMessages.ARENA_NOT_FOUND)
                    return@add
                }

                val event = Event(viewer, g.eventArenaId)
                event.start()
            }
        }
    }

    override val title: String
        get() = CC.BLUE + "Select Event"

    override fun shouldUpdate() = true
}
